src/NServiceBus.RabbitMQ/Receiving/MessagePump.cs

```diff
@@ -140,107 +139,77 @@ void Connection_ConnectionShutdown(object sender, ShutdownEventArgs e)
                 return;
             }
 
+            try
+            {
+                await ProcessMessage(eventArgs).ConfigureAwait(false);
+            }
+            finally
+            {
+                semaphore.Release();
+            }
+        }
+
+        private async Task ProcessMessage(BasicDeliverEventArgs message)
+        {
             Dictionary<string, string> headers = null;
             string messageId = null;
```

Comment:
Formatting: I'd prefer to have an empty line between the declarations and the `try`.